CurrencyConverterControl/CurrencyConverterControl.cs

```diff
@@ -0,0 +1,346 @@
+        /// <remarks>
+        /// The CurrencyList property represents a collection of Currency objects,
+        /// typically used for storing a list of currencies available in the application
+        /// to fill in the ComboBox element.
+        /// It is initialized with a new instance of ObservableCollection<Currency>.
+        /// </remarks>
+        [Description("Collection of currencies list")]
+        public ObservableCollection<Currency> CurrencyList { get; private set; } = new ObservableCollection<Currency>();
+
+        /// <summary>
+        /// Property to allow replacement of the currency converter
+        /// </summary>
+        public ICurrencyDataProvider CurrencyConverter
+        {
+            get { return CurrencyDataProvider; }
+            set { CurrencyDataProvider = value ?? throw new ArgumentNullException(nameof(value)); }
+        }
+
+        public static readonly DependencyProperty CustomStyleProperty =
+            DependencyProperty.Register(nameof(CustomStyle), typeof(Style), typeof(CurrencyConverterControl));
+        /// <summary>
+        /// Define custom styles on the control
+        /// </summary>
+        public Style CustomStyle
+        {
+            get { return (Style)GetValue(CustomStyleProperty); }
+            set { SetValue(CustomStyleProperty, value); }
+        }
+        #endregion
+        public override void OnApplyTemplate()
+        {
+
+            LoadCurrencies();
+            CurrencyDataProvider.GetConvertionRates();
+            // Check if existing controls are not null and unsubscribe from events
+            if (_currencyDestination != null)
+            {
+                _currencyDestination.Loaded -= ComboBox_Loaded;
+                _currencyDestination.SelectionChanged -= CmbDestination_SelectionChanged;
+            }
+
+            if (_currencySource != null)
+            {
+                _currencySource.Loaded -= ComboBox_Loaded;
+                _currencySource.SelectionChanged -= CmbSource_SelectionChanged;
+            }
+            _currencyDestination = Template.FindName("CmbDestination", this) as ComboBox;
+            _currencySource = Template.FindName("CmbSource", this) as ComboBox;
+            //To Be done
+            if (_currencyDestination is null || _currencySource is null)
+            { throw new InvalidOperationException("The Combobox in the Template is not found! Please double check!"); }
+            else 
+            {
+                _currencyDestination.Loaded += ComboBox_Loaded;         
+                _currencySource.Loaded += ComboBox_Loaded;
+                _currencyDestination.SelectionChanged += CmbDestination_SelectionChanged;
+                _currencySource.SelectionChanged += CmbSource_SelectionChanged;   
+            }
+                     
+            InputValue = 233.3; //providing a default value            
+            base.OnApplyTemplate();
+        }
+
+        public virtual Task LoadAsync()
```

Comment:
If this is only for testing, make it internal. I don't think this method is really needed in a public API.

See about InternalVisibleToAttribute:
https://learn.microsoft.com/en-US/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-6.0